src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClientV2.java

```diff
@@ -0,0 +1,114 @@
+package com.google.jenkins.plugins.computeengine.client;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Instance;
+import com.google.api.services.compute.model.InstancesScopedList;
+import com.google.api.services.compute.model.InstancesSetLabelsRequest;
+import com.google.cloud.graphite.platforms.plugin.client.ComputeClient;
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+/**
+ * Extends {@link com.google.cloud.graphite.platforms.plugin.client.ComputeClient} with additional functionalities.
+ * <p>This class serves as a venue for implementing features not available in the archived Graphite Java library
+ * (<a href="https://github.com/GoogleCloudPlatform/gcp-plugin-core-java">gcp-plugin-core-java</a>, last updated in December 2019).
+ * Consideration for the gradual evolution of this class is suggested, including the re-implementation of methods
+ * currently utilized from the Graphite library, to ensure dependency solely on the Google API Java Client Services
+ * library (<a href="https://github.com/googleapis/google-api-java-client-services">google-api-java-client-services</a>).
+ * This approach aims to eventually eliminate the reliance on the Graphite library.
+ */
+public class ComputeClientV2 {
+
+    private final String projectId;
+
+    @Getter
+    private final Compute compute;
+
+    public ComputeClientV2(String projectId, Compute compute) {
+        this.projectId = projectId;
+        this.compute = compute;
+    }
+
+    /**
+     * Instantiates a {@link ComputeClientV2} using the configuration from a given {@link ComputeEngineCloud}.
+     * <p>Jenkins may host multiple cloud configurations, each with distinct service accounts. This diversity necessitates leveraging an
+     * existing {@link ComputeEngineCloud} instance to accurately configure {@link ComputeClientV2}. Interfacing directly with
+     * {@link ComputeClient} is impractical due to its archived status and inability to extend, as noted in class-level documentation.
+     * Consequently, this method employs reflection to retrieve the necessary {@link Compute} instance from within {@link ComputeClient}.
```

Comment:
I do not think you actually need to do this at all. Unpack what `ClientFactory.<init>` and `.computeClient` and `.initializeRequest` actually does. It is not a lot. This plugin already supplies the `httpRequestInitializer` (from `getGoogleCredential`). `transport` is always empty, so you could use `GoogleNetHttpTransport.newTrustedTransport()` just like the Graphite version does. Same with `new JacksonFactory()`. Then the rest of the `Compute.Builder` setup is just setting the `applicationName`, and calling `setUserAgent` on `AbstractGoogleClientRequest`s.

Reply:
fixed now, it works, thank you